File: shopzz_flutter_app/lib/entity/response/coupon_response.dart
class CouponResponse {
  List<CouponEntity> couponList;

  CouponResponse({this.couponList});

  CouponResponse.fromJson(Map<String, dynamic> json) {
    if (json['couponList'] != null) {
      couponList = <CouponEntity>[];
      json['couponList'].forEach((v) {
        couponList.add(CouponEntity.fromJson(v));
      });
    }
  }

  Map<String, dynamic> toJson() {
    final Map<String, dynamic> data = <String, dynamic>{};
    if (couponList != null) {
      data['couponList'] = couponList.map((v) => v.toJson()).toList();
    }
    return data;
  }
}

class CouponEntity {
  int id;
  String name;
  String subName;
  String startTime;
  String endTime;
  num useThreshold;
  num discountAmount;
  num discountRate;
  int type;
  int fullLimited;

  CouponEntity(
      {this.id,
        this.name,
        this.subName,
        this.startTime,
        this.endTime,
        this.useThreshold,
        this.discountAmount,
        this.discountRate,
        this.type,
        this.fullLimited});

  CouponEntity.fromJson(Map<String, dynamic> json) {
    id = json['id'];
    name = json['name'];
    subName = json['subName'];
    startTime = json['startTime'];
    endTime = json['endTime'];
    useThreshold = json['useThreshold'];
    discountAmount = json['discountAmount'];
    discountRate = json['discountRate'];
    type = json['type'];
    fullLimited = json['fullLimited'];
  }

  Map<String, dynamic> toJson() {
    final Map<String, dynamic> data = <String, dynamic>{};
    data['id'] = id;
    data['name'] = name;
    data['subName'] = subName;
    data['startTime'] = startTime;
    data['endTime'] = endTime;
    data['useThreshold'] = useThreshold;
    data['discountAmount'] = discountAmount;
    data['discountRate'] = discountRate;
    data['type'] = type;
    data['fullLimited'] = fullLimited;
    return data;
  }
}

File: taowu-server/taowu-marketing/taowu-marketing-web/src/main/java/com/whoiszxl/taowu/cqrs/response/CouponResponse.java
package com.whoiszxl.taowu.cqrs.response;

import com.fasterxml.jackson.annotation.JsonFormat;
import io.swagger.v3.oas.annotations.media.Schema;
import lombok.Data;

import java.io.Serializable;
import java.math.BigDecimal;
import java.time.LocalDateTime;

/**
 * <p>
 * 优惠券表
 * </p>
 *
 * @author whoiszxl
 * @since 2022-03-29
 */
@Data
@Schema(description = "优惠券表")
public class CouponResponse implements Serializable {

    private static final long serialVersionUID = 1L;

    @Schema(description = "自增主键ID")
    private Long id;

    @Schema(description = "优惠券名称")
    private String name;

    @Schema(description = "优惠券副名称")
    private String subName;

    @Schema(description = "优惠券有效期起始时间")
    @JsonFormat(pattern = "yyyy-MM-dd HH:mm:ss", timezone = "GMT+8")
    private LocalDateTime startTime;

    @Schema(description = "优惠券有效期结束时间")
    @JsonFormat(pattern = "yyyy-MM-dd HH:mm:ss", timezone = "GMT+8")
    private LocalDateTime endTime;

    @Schema(description = "优惠券使用阈值")
    private BigDecimal useThreshold;

    @Schema(description = "折扣金额")
    private BigDecimal discountAmount;

    @Schema(description = "折扣比率")
    private BigDecimal discountRate;

    @Schema(description = "优惠券类型 1: 满减券, 2: 满减折扣券 3: 无门槛立减券")
    private Integer type;

    @Schema(description = "是否有全场限制 1: 全场通用, 2: 分类限制")
    private Integer fullLimited;

    @Schema(description = "优惠券状态 1: 有效, 2: 失效(超出有效期), 3: 系统停用")
    private Integer status;

    @Schema(description = "创建者")
    private String createdBy;

    @Schema(description = "更新者")
    private String updatedBy;

    @Schema(description = "创建时间")
    @JsonFormat(pattern = "yyyy-MM-dd HH:mm:ss", timezone = "GMT+8")
    private LocalDateTime createdAt;

    @Schema(description = "更新时间")
    @JsonFormat(pattern = "yyyy-MM-dd HH:mm:ss", timezone = "GMT+8")
    private LocalDateTime updatedAt;


}
